source/gameengine/Ketsji/KX_KetsjiEngine.cpp

```diff
@@ -1744,34 +1745,27 @@ bool KX_KetsjiEngine::GetShowArmatures() const
 	return m_showArmature;
 }
 
-void KX_KetsjiEngine::DrawCameraFrustum(KX_Camera *cam, KX_Scene *scene)
+void KX_KetsjiEngine::DrawCameraFrustum(KX_Camera *cam, KX_Scene *scene, const RAS_Rect& viewport)
 {
 	bool perspective = cam->GetCameraData()->m_perspective;
 
-	Scene *sc = scene->GetBlenderScene();
-	float aspx = (float)sc->r.xsch * sc->r.xasp;
-	float aspy = (float)sc->r.ysch * sc->r.yasp;
-	float ratiox = min_ff(aspx / aspy, 1.0f);
-	float ratioy = min_ff(aspy / aspx, 1.0f);
```

Comment:
Yes, my behaviour is wrong here, I noted the issue and maybe a little fix proposal in the framapad:
```
    It's pretty hard to draw the culling box of a camera which doesn't been used for culling (we don't know what the viewport/camera ratio will look like).

    I'm proposing to be able to render the scene from an other camera than the one used for culling and then be able to draw the frustum box of this culling camera. Is it correct ?
```